documentation/guide-batch-layer.asciidoc

```diff
@@ -212,8 +212,70 @@ Show help::
 As you can the each invocation includes the jdbc connection string to your database.
 This means that you have to make sure that the corresponding DB driver is in the classpath (the prepared jar only contains H2).
 
+== Authentication
+
+Most business application incorporate authentication and autorisation.
+Your spring boot application will implement some kind of security, e.g. integrated login with username+password or in many cases authentication via an existing IAM.
+For security reasons your batch should also implement an authentication mechanism and obey the authorization implemented in your application (e.g. via @RolesAllowed).
+
+Since there are many different authentication mechanism we cannot provide an out-of-the-box solution in devonfw, but we describe a pattern who this can be implemeneted in devonfw batches.
+
+We suggest to implement the authentication in a Spring Batch Tasklet, which runs as the first step in your batch. This tasklet will do all of the work which is required to authenticate the batch. A simple example which authenticates the batch "locally" via username and password could be implemente like this:
+
+[source,java]
+----
+@Named
+public class SimpleAuthenticationTasklet implements Tasklet {
+
+  @Override
+  public RepeatStatus execute(StepContribution contribution, ChunkContext chunkContext) throws Exception {
+
+    String username = chunkContext.getStepContext().getStepExecution().getJobParameters().getString("username");
+    String password = chunkContext.getStepContext().getStepExecution().getJobParameters().getString("password");
+    Authentication authentication = new UsernamePasswordAuthenticationToken(username, password);
+
+    SecurityContextHolder.getContext().setAuthentication(authentication);
```

Comment:
wont this overrule spring-security so the password is never checked? Or is there some magic in spring-batch, I am not aware of?

Reply:
This is Spring Security magic. Because Authentication.isAuthenticated is `false` Spring automatically does an authentication with the provided credentials.